src/nvim/decoration.h

```diff
@@ -48,6 +48,7 @@ typedef struct {
   int attr_id;  ///< cached lookup of inl.hl_id if it was a highlight
   bool owned;   ///< ephemeral decoration, free memory immediately
   DecorPriority priority;
+  DecorPriority subpriority;  ///< Secondary priority value used for ordering (#27131)
```

Comment:
can we also add another line here with @gpanders insight:

    /// ...defined by the order of patterns/captures in the query file.